include/tiny-cuda-nn/encoding.h

```diff
@@ -72,6 +72,17 @@ class Encoding : public DifferentiableObject<float, T, T> {
 	size_t n_params() const override { return 0; }
 
 	std::vector<std::pair<uint32_t, uint32_t>> layer_sizes() const override { return {}; }
+
+	void backward_backward_input(
```

Comment:
I'd prefer if `DifferentiableObject::backward_backward_input` already threw the exception. That'd save the trouble of overriding it in `Network` and `Encoding`.